_posts/2020-06-05-current-state-of-distributed-dask-clusters.md

```diff
@@ -0,0 +1,338 @@
+
+### Dask Helm chart
+
+Another option for running Dask on a Kubernetes cluster is using the [Dask Helm Chart](https://github.com/dask/helm-chart).
+
+This is an example of a fixed cluster setup. Helm is a way of installing specific resources on a Kubernetes cluster, similar to a package manager like `apt` or `yum`. The Dask Helm chart includes a Jupyter notebook, a Dask scheduler and three Dask workers. The workers can be scaled manually by interacting with the Kubernetes API but not adaptively by the Dask scheduler itself.
+
+This feels like a different approach to what we've seen so far. It gives you a Dask cluster which is always available, and a Jupyter notebook to drive the cluster from. You then have to take your work to the cluster's Jupyter session rather than spawning a cluster from your existing work place.
+
+One benefit of this approach is that because the Jupyter notebook has been set up as part of the cluster it already has the Lab Extension installed and also has been [pre-configured](https://github.com/dask/helm-chart/blob/f413647f90d6e278515b172c623977578a535aa2/dask/templates/dask-jupyter-deployment.yaml#L47-L48) with the location of the Dask cluster. So unlike previous examples where you need to either give the `Client` the address of the scheduler or a `Cluster` object, in this instance it will auto-detect the cluster from environment variables that are set by the Helm chart.
+
+```python
+from dask.distributed import Client
+
+client = Client()  # The address is loaded from an environment variable
+```
+
+_**Note:** If you call `Client` without any arguments in other situations where the scheduler location has not been configured it will automatically create a `LocalCluster` object and use that._
+
+### Dask Jobqueue
+
+[Dask Jobqueue](https://github.com/dask/dask-jobqueue) is a set of cluster managers aimed at HPC users.
+
+When working as a researcher or academic with access to an HPC or Supercomputer you likely have to submit work to that machine via some kind of job queueing system. This is often in the form of a bash script which contains metadata about how much resource you need on the machine and the commands you want to run.
+
+Dask Jobqueue has cluster manager objects for [PBS](https://en.wikipedia.org/wiki/Portable_Batch_System), [Slurm](https://en.wikipedia.org/wiki/Slurm_Workload_Manager), and [SGE](https://en.wikipedia.org/wiki/Oracle_Grid_Engine). When creating these cluster managers they will construct scripts for the batch scheduler based on your arguments and submit them using your default credentials.
+
+```python
+from dask.distributed import Client
+from dask_jobqueue import PBSCluster
+
+cluster = PBSCluster()
```

Comment:
Since this is an overview of quite a few types of Dask clusters, I agree that you probably do not want to go into the details for each Dask cluster, but I am wondering how to best convey that this is not runnable code (unless you happen to have modified your configuration file). People may try to copy and paste this and be annoyed if they get an error

Maybe something like:
```py
cluster = PBSCluster(**some_cluster_specific_arguments_go_here)
```

I am guessing that `FooCluster()` does not work for other Dask cluster classes in this blog post but maybe I am wrong.

Reply:
Yeah good point. This is true for most other cluster classes. 

Perhaps something like `**some_cluster_specific_arguments_go_here` as you suggest plus a link to the docs for each cluster manager would make sense?

Reply:
> Perhaps something like `**some_cluster_specific_arguments_go_here` as you suggest plus a link to the docs for each cluster manager would make sense?

Seems reasonable.